pulpcore/cli/common/generic.py

```diff
@@ -42,15 +51,58 @@ def format_help_text(
             self.help = self.help.format(entity=entity_ctx.ENTITY, entities=entity_ctx.ENTITIES)
         super().format_help_text(ctx, formatter)
 
+    def get_params(self, ctx: click.Context) -> List[click.Parameter]:
+        params = super().get_params(ctx)
+        new_params: List[click.Parameter] = []
+        # Is it more efficient to go through and just remove options from params?
+        for param in params:
+            if isinstance(param, PulpOption):
+                if param.needs_contexts is not None:
+                    if not isinstance(ctx.obj, param.needs_contexts):
+                        continue
+            new_params.append(param)
+        return new_params
+
+
+class PulpGroup(click.Group):
+    def command(self, *args: Any, **kwargs: Any) -> Callable[[_FC], click.Command]:
```

Comment:
I figured, we can be more specific on the type definitions used here.
Can you replace them?
```
_F = Callable[..., Any]
_FC = TypeVar("_FC", _F, click.Command)